### PR TITLE
Added basic config builder

### PR DESCRIPTION
**Pull request by aluttik**
The readme will still need more work and it wouldn't hurt to add a few more tests for this, but the basics are all there.

```
usage: crossplane build [-h] [-v] [-d PATH] [-f] [-i NUM | -t] [--no-headers]
                        [--stdout]
                        filename

builds an nginx config from a json payload

positional arguments:
  filename              the file with the config payload

optional arguments:
  -h, --help            show this help message and exit
  -v, --verbose         verbose output
  -d PATH, --dir PATH   the base directory to build in
  -f, --force           overwrite existing files
  -i NUM, --indent NUM  number of spaces to indent output
  -t, --tabs            indent with tabs instead of spaces
  --no-headers          do not write header to configs
  --stdout              write configs to stdout instead
```

### CODE REVIEW

File: crossplane/__main__.py [hunk 1]
```diff
@@ -86,6 +43,59 @@ def callback(e):
...
+    dirs_to_make = []
+
+    for config in payload['config']:
+        path = os.path.join(dirname, config['file'])
+        dirpath = os.path.dirname(path)
+        if os.path.exists(path):
+            existing.append(path)
+        elif not os.path.exists(dirpath) and dirpath not in dirs_to_make:
+            dirs_to_make.append(dirpath)
+
```
Comment (aluttik) [inline crossplane/__main__.py]: this could probably use some inline documentation

File: crossplane/__main__.py [hunk 2]
```diff
@@ -86,6 +43,59 @@ def callback(e):
...
+    for dirpath in dirs_to_make:
+        os.makedirs(dirpath)
+
+    for config in payload['config']:
+        path = os.path.join(dirname, config['file'])
+
+        if header:
+            output = (
+                '# This config was built from JSON using crossplane.\n'
```
Comment (dedm) [inline crossplane/__main__.py]: NGINX crossplane (branding!)

Comment (aluttik) [reply crossplane/__main__.py]: haha okay

File: crossplane/builder.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    arg = str(arg.encode('utf-8') if PY2 else arg)
+    if _needs_quotes(arg):
+        arg = repr(arg.decode('string_escape') if PY2 else arg)
+    return arg
+
+
+def build(payload, indent=4, tabs=False):
+    padding = '\t' if tabs else ' ' * indent
+
+    def _build_lines(objs, depth):
```
Comment (gshulegaard) [inline crossplane/builder.py]: This is a really clever use of a generator.

Comment (aluttik) [reply crossplane/builder.py]: Thanks :)

**Pull request merged**